rollout/trafficrouting/istio/istio.go

```diff
@@ -765,13 +789,13 @@ func (r *Reconciler) generateHeaderBasedPatches(httpRoutes []VirtualServiceHTTPR
 	}
 
 	patches := virtualServiceRoutePatches{}
-	headerRouteExist := hasHeaderRoute(httpRoutes)
+	index := getHeaderRouteIndex(headerRouting, httpRoutes)
 
-	if headerRouteExist {
+	if index >= 0 {
 		if headerRouting == nil || headerRouting.Match == nil {
-			deleteHeaderRoute(&patches)
+			deleteHeaderRoute(index, &patches)
```

Comment:
I think we should have a test case to validate this flow.

Reply:
I can just remove all that code instead.